patches/minecraft/net/minecraft/world/level/block/JukeboxBlock.java.patch

```diff
@@ -0,0 +1,23 @@
+--- a/net/minecraft/world/level/block/JukeboxBlock.java
++++ b/net/minecraft/world/level/block/JukeboxBlock.java
+@@ -27,7 +_,7 @@
+ import net.minecraft.world.level.gameevent.GameEvent;
+ import net.minecraft.world.phys.BlockHitResult;
+ 
+-public class JukeboxBlock extends BaseEntityBlock {
++public class JukeboxBlock extends BaseEntityBlock implements net.minecraftforge.common.extensions.IForgeJukebox {
+    public static final BooleanProperty f_54254_ = BlockStateProperties.f_61439_;
+ 
+    public JukeboxBlock(BlockBehaviour.Properties p_54257_) {
+@@ -55,6 +_,11 @@
+       } else {
+          return InteractionResult.PASS;
+       }
++   }
++
++   @Override
++   public void insertRecord(@org.jetbrains.annotations.Nullable Entity entity, LevelAccessor level, BlockPos pos, BlockState state, ItemStack stack) {
```

Comment:
What is the purpose of this method? Given the fact that `m_238345_` isn't final why not just have people override that instead?

Reply:
In the forge discord, it was recommended to create an extension to abstract the functionality so blocks not extending the vanilla Jukebox can work with records. The inserting of records are based on the record item and its hardcoded. The reason for this PR.

Reply:
At the very least then default impl should be done in the interface and instance check to see if it is a jukebox entity rather than having an extra patch to this file.